app/src/main/java/io/homeassistant/companion/android/settings/vehicle/views/AndroidAutoFavoritesView.kt

```diff
@@ -0,0 +1,117 @@
+import io.homeassistant.companion.android.settings.vehicle.ManageAndroidAutoViewModel
+import io.homeassistant.companion.android.util.compose.FavoriteEntityRow
+import io.homeassistant.companion.android.util.compose.ServerExposedDropdownMenu
+import io.homeassistant.companion.android.util.compose.SingleEntityPicker
+import org.burnoutcrew.reorderable.ReorderableItem
+import org.burnoutcrew.reorderable.rememberReorderableLazyListState
+import org.burnoutcrew.reorderable.reorderable
+import io.homeassistant.companion.android.common.R as commonR
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun AndroidAutoFavoritesSettings(
+    androidAutoViewModel: ManageAndroidAutoViewModel,
+    serversList: List<Server>,
+    defaultServer: Int
+) {
+    val reorderState = rememberReorderableLazyListState(
+        onMove = { from, to -> androidAutoViewModel.onMove(from, to) },
+        canDragOver = { draggedOver, _ -> androidAutoViewModel.canDragOver(draggedOver) },
+        onDragEnd = { _, _ -> androidAutoViewModel.saveFavorites() }
+    )
+
+    var selectedServer by remember { mutableStateOf(defaultServer) }
+
+    val favoriteEntities = androidAutoViewModel.favoritesList
+
+    LazyColumn(
+        state = reorderState.listState,
+        contentPadding = PaddingValues(vertical = 16.dp),
+        modifier = Modifier
+            .padding(16.dp)
+            .reorderable(reorderState)
+    ) {
+        item {
+            Text(
+                text = stringResource(commonR.string.aa_set_favorites),
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.padding(horizontal = 16.dp)
+            )
+        }
+
+        if (serversList.size > 1) {
+            item {
+                ServerExposedDropdownMenu(
+                    servers = serversList,
+                    current = selectedServer,
+                    onSelected = {
+                        androidAutoViewModel.loadEntities(it)
+                        selectedServer = it
+                    },
+                    modifier = Modifier.fillMaxWidth().padding(start = 16.dp, end = 16.dp, bottom = 16.dp)
+                )
+            }
+        }
+        item {
+            SingleEntityPicker(
+                entities = androidAutoViewModel.sortedEntities,
+                currentEntity = null,
+                onEntityCleared = { /* Nothing */ },
+                onEntitySelected = {
+                    androidAutoViewModel.onEntitySelected(true, it, selectedServer)
+                    return@SingleEntityPicker false // Clear input
+                },
+                modifier = Modifier.padding(all = 16.dp),
+                label = { Text(stringResource(commonR.string.add_favorite)) }
+            )
+        }
+        if (favoriteEntities.isNotEmpty() && androidAutoViewModel.sortedEntities.isNotEmpty()) {
+            items(favoriteEntities.size, { favoriteEntities[it] }) { index ->
```

Comment:
The driving interface only shows one server's favorites at a time, you switch servers to see the other server favorites. Shouldn't the same apply to the settings for favorites?

Reply:
done, we now account for the server ID